Circle's nightly has been broken

costinm opened the pull request:
Some files are missing, and it seems the release script was using the wrong dir ?

rshriram commented:
ack for circle changes

costinm commented on release/create_release_archives.sh:
```diff
@@ -107,8 +107,10 @@ function create_windows_archive() {
-pushd "${OUTPUT_PATH}"
-${CP} istio.VERSION LICENSE README.md "${COMMON_FILES_DIR}"/
+# Not sure why it was used before - but the files are in source, not out
+# pushd "${OUTPUT_PATH}"
```
Sounds good - but right now it doesn't seem to be working, and since updateVersion is mostly going away (most logic is in helm template) I don't think 'scribble the source while building the release' is a huge issue.

I think it's a larger issue that it's generating stuff in the source tree for normal development, where it affects a lot of people - but proper fix is in fixing updateVersion to generate in out, not in release script to work around.

ldemailly replied:
let's not break the official release process though

it makes sense it needs a "clean" git tree (for instance to test for cleanness)

is there a way to run a PR through the daily build process ?

the last one https://gcsweb.istio.io/gcs/istio-prerelease/daily-build/
is still 3/25 and I don't want that we continue breaking it - we need those builds for perf testing and for sanity checks and to follow our daily build process

cc on call @cmluciano

mattdelco replied:
> is there a way to run a PR through the daily build process ?

The closest way to do a test build is to commit changes to a forked repo of istio (e.g., "master4" branch of "mattdelco/istio"), create a repo-tool manifest file that points to that fork (e.g., I use a "mattdelco/imanifest" repo where I've basically got an xml file for each branch I use [which is partly why I tend to reuse branches with boring names] like  https://github.com/mattdelco/manifest/blob/master/master4.xml ) and then use my gcloud acct to request a Cloud Builder build that stores the results in GCR/GCS locations that belong to that account), e.g.:

<pre>P_ID=${USER}-experimental
SVC_ACCT="cloudbuild@${P_ID}.iam.gserviceaccount.com"

MFEST_URL=https://github.com/mattdelco/manifest
MFEST_FILE="master4.xml"
MFEST_COMMIT="master"
VERSION="0.7.0-20180322"

GCR_DEST="${P_ID}/prerelease"
GCS_DEST="delco-prerelease/prerelease/${VERSION}"

./release/start_gcb_build.sh -w -p "${P_ID}" -a "${SVC_ACCT}" -r "${GCR_DEST}" -s "${GCS_DEST}" \
  -v "${VERSION}" -u "${MFEST_URL}" -t "${MFEST_COMMIT}" -m "${MFEST_FILE}"
</pre>

If you've got a Linux machine with docker, fpm, and gcloud/gsutil installed then in theory you can also run the same steps of the build on your local system, e.g.:

<pre>SCRATCH_DIR=/tmp/output
GCLOUD_ACCT=${USER}-experimental
VER_STRING=0.7.0-20180328
GCS_PATH=${GCLOUD_ACCT}/testbuild/${VER_STRING}
GCR_PATH=${GCLOUD_ACCT}

mkdir -p ${SCRATCH_DIR}
./release/cloud_builder.sh -p ${GCS_PATH} -q ${GCR_PATH} -t ${VER_STRING} -o ${SCRATCH_DIR}
./release/store_artifacts.sh -p ${GCS_PATH} -q ${GCR_PATH} -v ${VER_STRING} -o {SCRATCH_DIR}
</pre>

If you don't want to bother with Google infra then it's probably sufficient to just run cloud_builder.sh (the script takes GCS/GCR paths as input but I think they're only used by that script [via updateVersion.sh] to bake the paths into the product.

ldemailly commented:
let's postpone this after #4642 and we have a way to test it

costinm commented:
Ok, will remove the 'release' stuff from the PR and leave only the circle.

Right now the release script is broken - this PR would make it work again, not break it.

istio-testing commented:
@costinm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e-v1alpha3.sh | b94b2c1f2f7d185acd8510809dd46fc90a1e8a23 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4639/istio-pilot-e2e-v1alpha3/503/) | `/test istio-pilot-e2e-v1alpha3`
prow/istio-pilot-e2e.sh | b94b2c1f2f7d185acd8510809dd46fc90a1e8a23 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4639/istio-pilot-e2e/5059/) | `/test istio-pilot-e2e`
prow/istio-unit-tests.sh | 4dcbda077c6034807dd602cf53fed3745b1e5086 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4639/istio-unit-tests/3370/) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@costinm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e-v1alpha3.sh | b94b2c1f2f7d185acd8510809dd46fc90a1e8a23 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4639/istio-pilot-e2e-v1alpha3/503/) | `/test istio-pilot-e2e-v1alpha3`
prow/istio-pilot-e2e.sh | b94b2c1f2f7d185acd8510809dd46fc90a1e8a23 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4639/istio-pilot-e2e/5059/) | `/test istio-pilot-e2e`
prow/istio-unit-tests.sh | 4dcbda077c6034807dd602cf53fed3745b1e5086 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4639/istio-unit-tests/3370/) | `/test istio-unit-tests`
prow/istio-presubmit.sh | 4dcbda077c6034807dd602cf53fed3745b1e5086 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4639/istio-presubmit/8075/) | `/test istio-presubmit`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: 
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **costinm**

Assign the PR to them by writing `/assign @costinm` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[.circleci/OWNERS](https://github.com/istio/istio/blob/master/.circleci/OWNERS)**
- **[release/OWNERS](https://github.com/istio/istio/blob/master/release/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["costinm"]} -->

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: 
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **rshriram**

Assign the PR to them by writing `/assign @rshriram` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[.circleci/OWNERS](https://github.com/istio/istio/blob/master/.circleci/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["rshriram"]} -->

costinm commented:
Ok, reverted the fixes to make archive/release - will let Mandar push his.

Just changed circle to do the make archive at the end and save it as artifact.

ldemailly commented on Makefile:
```diff
@@ -20,7 +20,7 @@ export ISTIO_GO
 
 # Current version, updated after a release.
-VERSION ?= 0.8.0
+VERSION ?= 0.5.0
```
?? what is this VERSION for btw ?

ldemailly commented on .circleci/config.yml:
```diff
@@ -624,14 +624,14 @@ jobs:
     resource_class: xlarge
     environment:
       TAG: circleci-nightly
+      VERSION: 0.8-dev
```
which pr/build made this version/what's different about it ?

costinm replied:
Default is 0.5.0 
Circle is building nightly versions of 0.8 - obviously 0.8.0. would be a bad name. 
Not sure what is the problem here - continuing to generate '0.5.0' doesn't seem right.

istio-testing commented:
@costinm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e.sh | b94b2c1f2f7d185acd8510809dd46fc90a1e8a23 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4639/istio-pilot-e2e/5059/) | `/test istio-pilot-e2e`
prow/istio-pilot-e2e-v1alpha3.sh | b926d0f117a63f92f799a08df5415d56f7b4848f | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4639/istio-pilot-e2e-v1alpha3/514/) | `/test istio-pilot-e2e-v1alpha3`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@costinm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e-v1alpha3.sh | b926d0f117a63f92f799a08df5415d56f7b4848f | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4639/istio-pilot-e2e-v1alpha3/514/) | `/test istio-pilot-e2e-v1alpha3`
prow/istio-pilot-e2e.sh | b926d0f117a63f92f799a08df5415d56f7b4848f | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4639/istio-pilot-e2e/5070/) | `/test istio-pilot-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->
